app/services/container_deployment_service.rb

```diff
@@ -51,4 +52,8 @@ def optional_vms(vms)
       }
     end
   end
+
+  def optional_deployment_types
+    ["origin", "openshift-enterprise"]
```

Comment:
:+1: for openshift-origin
formatting can be handled by clients but they should submit to the  server what they originaly got